Jint/Native/Function/ScriptFunctionInstance.cs

```diff
@@ -94,6 +95,16 @@ public override JsValue Call(JsValue thisArg, JsValue[] arguments)
 
                     if (result.Type == Completion.Throw)
                     {
+                        if (result.Value.HasValue && result.Value.Value.IsObject())
```

Comment:
It looks like that `TryCast` will cast `ObjectInstance` to `Exception` not object itself.